cmd/bpfman-agent/main.go

```diff
@@ -43,6 +43,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/metrics/server"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 	//+kubebuilder:scaffold:imports
+	_ "net/http/pprof"
```

Comment:
I think you can remove this import.
IIRC this adds the routes to the default ServeMux... 
However, you're passing the bind port through to controller-runtime, which doesn't use the default ServeMux...